coq/coq-system.el

```diff
@@ -378,12 +378,26 @@ options of a few coq-project files does the right thing."
 LOADPATH, CURRENT-DIRECTORY, PRE-V85: see `coq-include-options'."
   (coq-include-options loadpath current-directory pre-v85))
 
+(defun coq-clean-prog-args (args)
+  "Return a copy of ARGS with automatically added entries of `coq-prog-args' removed.
+
+Such entries are currently -emacs and -topfile."
+  (if (and (consp args) (equal (car args) "-emacs"))
```

Comment:
The condition `(consp args)` on this line and the subsequent ones could be factored-out, and you could use a `(case ...)`, for example like this:

```lisp
(defun coq-clean-prog-args (args)
  "Return a copy of ARGS with automatically added entries of `coq-prog-args' removed.
Such entries are currently -emacs and -topfile."
  (if (consp args)
      (cond
       ((equal (car args) "-emacs")
	(coq-clean-prog-args (cdr args)))
       ((and (consp (cdr args))
             (equal (car args) "-topfile")
             (equal (cadr args) buffer-file-name))
	(coq-clean-prog-args (cddr args)))
       (t
	(cons (car args) (coq-clean-prog-args (cdr args)))))
      args))
```

Reply:
I looked around and rewrote using pcase instead

Reply:
Thanks! it's even more readable.